File: src/main/java/szewek/mcflux/fluxcompat/ForgeEnergyCapable.java
package szewek.mcflux.fluxcompat;

import szewek.fl.energy.IEnergy;

public abstract class ForgeEnergyCapable implements IEnergy, net.minecraftforge.energy.IEnergyStorage {
	@Override public int receiveEnergy(int maxReceive, boolean simulate) {
		return (int) inputEnergy(maxReceive, simulate);
	}

	@Override public int extractEnergy(int maxExtract, boolean simulate) {
		return (int) outputEnergy(maxExtract, simulate);
	}

	@Override public int getEnergyStored() {
		long e = getEnergy();
		return e > Integer.MAX_VALUE ? Integer.MAX_VALUE : (int) e;
	}

	@Override public int getMaxEnergyStored() {
		long e = getEnergyCapacity();
		return e > Integer.MAX_VALUE ? Integer.MAX_VALUE : (int) e;
	}

	@Override public boolean canExtract() {
		return canOutputEnergy();
	}

	@Override public boolean canReceive() {
		return canInputEnergy();
	}
}

File: src/main/java/szewek/mcflux/fluxcompat/ForgeEnergyCapable.kt
package szewek.mcflux.fluxcompat

import szewek.fl.energy.IEnergy

abstract class ForgeEnergyCapable : IEnergy, net.minecraftforge.energy.IEnergyStorage {
	override fun receiveEnergy(maxReceive: Int, simulate: Boolean) =
			inputEnergy(maxReceive.toLong(), simulate).toInt()

	override fun extractEnergy(maxExtract: Int, simulate: Boolean) =
			outputEnergy(maxExtract.toLong(), simulate).toInt()

	override fun getEnergyStored(): Int {
		val e = energy
		return if (e > Integer.MAX_VALUE) Integer.MAX_VALUE else e.toInt()
	}

	override fun getMaxEnergyStored(): Int {
		val e = energyCapacity
		return if (e > Integer.MAX_VALUE) Integer.MAX_VALUE else e.toInt()
	}

	override fun canExtract() = canOutputEnergy()

	override fun canReceive() = canInputEnergy()
}
